# Problem Set 4

11/28/2022

## 1. Calculate total emittance of a blackbody radiator at 900 K. Calculate wavelength of peak emitted radiance

In [85]:
import numpy as np
import pandas as pd
import math

## a) total emittance across all wavelengths (Stefan-Boltzmann)

In [86]:
sigma = 5.67e-8
T = 900

In [87]:
M = sigma * T**4
M

37200.869999999995

Total emittance is 37201 W/m2

### b) wavelength of peak emittance (Wien's)

In [88]:
wiens = 2.8978e-3 #units mK

In [89]:
lambda_max = wiens / T
lambda_max 

3.2197777777777777e-06

Wavelength of peak emittance is 3.22 um

## Problem 2

Emissivity of quartis is approx 0.3 at 9 um. Given kinetic temp of 287 K, calculaetd emitted spectral radiance and brightness temperature of quartz at this wavelength

In [90]:
T_k = 287 #Kelvin
e = 0.3
lambda_2 = 9e-6 #meters
h = 6.626e-34 # Joule seconds
c = 3e8 # meters / second
k = 1.38e-23 #Joule / Kelvin

In [91]:
def calc_emitted_spectral_radiance(e, h, c, k, wavelength, Tk):
    
    L = e*((2*h*c**2)/((wavelength**5) * (math.exp((h*c)/(k*wavelength*Tk)) - 1)))
    return L

In [92]:
L = calc_emitted_spectral_radiance(0.3, h, c, k, lambda_2, T_k)

In [93]:
L

2302474.471191069

Now, calculate brightness temp

In [43]:
def calc_brightness_temp(h, c, k, wavelength, Tk, e):
    
    Tb = (h*c)/(k*wavelength*math.log((math.exp((h*c)/(k*wavelength*Tk)) + e - 1)/e))
    return Tb

In [44]:
Tb = calc_brightness_temp(h,c,k,lambda_2, T_k, e)
Tb

236.13204709450392

## Problem 3

Measured forest radiance is 0.007 Wm-2sr-1nm-1 at 10 um. Emissivity of forest is 0.95, find kinetic temperature

In [45]:
lambda_3 = 1000e-9
L_3 = 0.007e-9
e_3 = 0.95

In [46]:
def calc_kinetic_temp(h,c,k,wavelength,e, L):
    
    Tk = (h*c)/(k*wavelength*math.log(((e*2*h*c**2)/(L*wavelength**5))+1))
    return Tk

In [47]:
Tk = calc_kinetic_temp(h,c,k,lambda_3, e_3, L_3)
Tk

248.1531071013427

Kinetic temp is 248 Kelvin

In [36]:
T_k3 = (h*c) / (k*lambda_3*math.log(1 + ((e_3 * 2 * h * c**2) / (L_3 * lambda_3**5))))

In [37]:
T_k3

248.1531071013427

## Problem 4

In [52]:
def calc_kinetic_temp_given_Tb(h,c,k,wavelength, e, Tb):
    
    Tk = (h*c)/(k*wavelength*math.log(1+e*math.exp((h*c)/(k*wavelength*Tb))-e))
    return Tk

In [53]:
Tb4_rock = 302 #kelvin
Tb4_mix = 778 #kelvin
e4_lava = 0.97
lambda_4 = 4e-6 #meters

In [116]:
## new verison   #def calc_emitted_spectral_radiance(e, h, c, k, wavelength, Tk):
L_rock = calc_emitted_spectral_radiance(1, h,c,k, lambda_4, Tb4_rock)
L_rock

772046.4576812931

In [117]:
L_mix = calc_emitted_spectral_radiance(1, h,c,k, lambda_4, Tb4_mix)
L_mix

1148924230.0475092

In [118]:
quantity_lava = L_mix - (0.68 * L_rock)

In [119]:
L_lava = quantity_lava / 0.38
L_lava

3022103259.0954895

In [120]:
Tk_lava = calc_kinetic_temp(h,c,k,lambda_4, e4_lava, L_lava)

In [121]:
Tk_lava

987.2369057499196